### PR TITLE
Adds campfires

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

Adds campfires that provide heating to those nearby, while requiring fuel

Sprites and sound by Thwomper
# Explain why it's good for the game

They look cool and will allow marines, and maps, to have campfires without using braziers

# Changelog

:cl:
add: Adds campfires, craftable with wood
code: Modifies brazier construction code for easier use and mapping
/:cl:

### CODE REVIEW

**Comment by BeagleGaming1** [comment]
I'll fix the maps later

File: code/game/objects/structures/props.dm [hunk 1]
```diff
@@ -695,25 +717,136 @@
...
+	frame_type = /obj/structure/prop/brazier/campfire
+	density = FALSE
+
+/obj/structure/prop/brazier/frame/full/campfire/smolder
+	name = "smoldering campfire"
+	desc = "A campfire that used to be lit, but was extinguished. You can still see the embers, and smoke rises from it."
+	state = STATE_FUEL
+	frame_type = /obj/structure/prop/brazier/frame/full/campfire
+
+/obj/structure/prop/brazier/campfire
```
Comment (BeagleGaming1) [inline code/game/objects/structures/props.dm:745]: I'm considering moving this code up to `/obj/structure/prop/brazier`

File: code/game/objects/structures/props.dm [hunk 2]
```diff
@@ -695,25 +717,136 @@
...
+		time_to_sound = initial(time_to_sound)
+
+/obj/structure/prop/brazier/campfire/attack_hand(mob/user)
+	. = ..()
+	if(!fuel_stage_time)
+		to_chat(user, SPAN_WARNING("You cannot extinguish [src]."))
+		return
+	to_chat(user, SPAN_NOTICE("You begin to extinguish [src]."))
+	while(remaining_fuel)
+		if(!do_after(user, 3 SECONDS, INTERRUPT_MOVED, BUSY_ICON_BUILD))
```
Comment (harryob) [inline code/game/objects/structures/props.dm:800]:
```suggestion
		if(user.action_busy || !do_after(user, 3 SECONDS, INTERRUPT_MOVED, BUSY_ICON_BUILD))
```

File: code/game/objects/structures/props.dm [hunk 3]
```diff
@@ -695,25 +717,136 @@
...
+	visible_message(SPAN_NOTICE("[user] fuels [src] with [fuel]."))
+	remaining_fuel++
+
+/obj/structure/prop/brazier/campfire/attack_alien(mob/living/carbon/xenomorph/xeno)
+	if(!fuel_stage_time)
+		to_chat(xeno, SPAN_WARNING("You cannot extinguish [src]."))
+		return
+	to_chat(xeno, SPAN_NOTICE("You begin to extinguish [src]."))
+	while(remaining_fuel)
+		if(!do_after(xeno, 1 SECONDS, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
```
Comment (harryob) [inline code/game/objects/structures/props.dm:824]:
```suggestion
		if(xeno.action_busy || !do_after(xeno, 1 SECONDS, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by FransFeiffer — APPROVED** [review]
Sprites look nice, here's a spriter approval 👍

**Review by harryob — APPROVED** [review]

**Pull request merged**